### PR TITLE
[TypeHints] `OneHotDegree`

### PR DESCRIPTION
**Pull request by sebastian-montero**

### CODE REVIEW

**Comment by sebastian-montero** [comment]
TorchScript still doesn't support variable list of arguments with *args
```
E           torch.jit.frontend.NotSupportedError: Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults:
E             File "/Users/sebastianmontero/Desktop/pytorch_geometric/torch_geometric/data/data.py", line 735
E               def __call__(self, *args: List[str]) -> Iterable:
E                                  ~~~~~ <--- HERE
E                   r"""Iterates over all attributes :obj:`*args` in the data, yielding
E                   their attribute names and values.
```

File: torch_geometric/transforms/one_hot_degree.py [hunk 1]
```diff
@@ -19,12 +22,14 @@ class OneHotDegree(BaseTransform):
...
-    def __init__(self, max_degree, in_degree=False, cat=True):
+    def __init__(self, max_degree: int, in_degree: bool = False,
+                 cat: bool = True):
         self.max_degree = max_degree
         self.in_degree = in_degree
         self.cat = cat
 
-    def __call__(self, data):
+    def __call__(self, data: Union[Data,
+                                   HeteroData]) -> Union[Data, HeteroData]:
```
Comment (EdisonLeeeee) [inline torch_geometric/transforms/one_hot_degree.py:32]: `HeteroData` is not supported for this method.

Comment (rusty1s) [inline torch_geometric/transforms/one_hot_degree.py:32]:
```suggestion
    def __call__(self, data: Data) -> Data:
```

**Review by rusty1s — APPROVED** [review]
Thank you!

**Pull request merged**